src/HotChocolate/Data/src/Data/Filters/Types/RelayIdFilterFieldExtensions.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using HotChocolate.Configuration;
+using HotChocolate.Internal;
+using HotChocolate.Types;
+using HotChocolate.Types.Descriptors;
+using HotChocolate.Types.Relay;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace HotChocolate.Data.Filters
+{
+    internal static class RelayIdFilterFieldExtensions
+    {
+        public static IFilterOperationFieldDescriptor ID<TEntityType, TIdType>(
+            this IFilterOperationFieldDescriptor descriptor)
+        {
+            if (descriptor is null)
+            {
+                throw new ArgumentNullException(nameof(descriptor));
+            }
+
+            descriptor.Extend()
+                .OnBeforeCompletion(
+                    (c, d) => AddSerializerToInputField<TEntityType, TIdType>(c, d));
+
+            return descriptor;
+        }
+
+        private static void AddSerializerToInputField<TEntityType, TIdType>(
+            ITypeCompletionContext completionContext,
+            FilterOperationFieldDefinition definition)
+        {
+            ITypeInspector typeInspector = completionContext.TypeInspector;
+            IExtendedType? resultType;
+
+            resultType = typeInspector.GetReturnType(typeof(TIdType[]), true);
```

Comment:
This works but I am not sure if it is a hack
Without this it ListValues are not supported.